src/properties.c

```diff
@@ -714,24 +829,36 @@ int properties_update_data_item(struct knot_thing *thing, char *filename,
 	/* Update values on knot_thing struct */
 	device_update_config_data_item(thing, config);
 
+	has_err = false;
 	data_item_group = get_data_item_groups(device_fd);
 
 	for (i = 0; data_item_group[i] != NULL; i++) {
 		if (!equal_sensor_id(device_fd, data_item_group[i],
 				     config->sensor_id))
 			continue;
 
-		update_schema_data_item(device_fd, &config->schema);
-
-		if (!config->event.event_flags & KNOT_EVT_FLAG_UNREGISTERED)
-			update_event_data_item(device_fd,
-					       config->schema.value_type,
-					       &config->event);
+		if (update_schema_data_item(device_fd, data_item_group[i],
+					    &config->schema) < 0) {
+			has_err = true;
+			l_error("Error on set schema property");
+		}
 
+		if (!(config->event.event_flags & KNOT_EVT_FLAG_UNREGISTERED)) {
+			if (update_event_data_item(device_fd,
```

Comment:
If this validation purpose is to check if it is only UNREGISTERED, I believe that this way works because KNOT_EVT_FLAG values were defined to not occur bit superposition.

That is, the only case that '&' operator will return a true value is when event_flags has the UNREGISTERED_FLAG, in all the other cases it will  return false.

Reply:
I think that the @alla3 comment is about to unify the ifs with && operator. But I think that is unnecessary, because will decrease the code readability and will not increase the performace.